starcli/search.py

```diff
@@ -226,15 +226,17 @@ def search(
 
 
 def search_github_trending(
-    language=None, spoken_language=None, order="desc", stars=">=10", date_range=None
+    languages=[""], spoken_language=None, order="desc", stars=">=10", date_range=None
 ):
     """Returns trending repositories from github trending page"""
-    if date_range:
-        gtrending_repo_list = fetch_repos(
-            language, spoken_language, date_range_map[date_range]
-        )
-    else:
-        gtrending_repo_list = fetch_repos(language, spoken_language)
+    gtrending_repo_list = []
+    for language in languages:
```

Comment:
This is rather slow at 3+ languages. I wish `gtrending` would allow for supplying multiple languages. Caching alleviates the problem a bit though. However, it takes just changing the order of langs or adding another one to invalidate the cache entry. Possibly we should consider caching at a lower level here as well.

Reply:
| Perhaps a set could be used when comparing the language parameter of cache and current query?
Yes, if the cache layer for gtrending queries would be implemented. I guess it's kinda similar problem as with GQL cache, where set is often leveraged as well. 

| Sorry, what do you mean?
I was considering caching these per-programming-language gtrending queries so that we can re-use them if the original multi-lang query had programming language added/removed

I agree the current implementation is good enough.

I noticed that GitHub now has a page with trending repos, where you can select spoken language. However, this feature is not (yet) exposed to GH API for whatever weird reason. Assuming this will be ported to the API at some point in the future, we'll be able to get rid of gtrending and leverage GH API exclusively.